catalog/view/theme/default/template/extension/payment/securesubmit.tpl

```diff
@@ -1,156 +1,327 @@
-    <!--
-            $(document).ready(function () {
-                $('#button-confirm').bind('click', secureSubmitFormHandler);
-                $("#input-cc-number").keydown(function (e) {
-                    // Allow: backspace, delete, tab, escape, enter and .
-                    if ($.inArray(e.keyCode, [46, 8, 9, 27, 13, 110, 190]) !== -1 ||
-                                // Allow: Ctrl+A
-                            (e.keyCode == 65 && e.ctrlKey === true) ||
-                                // Allow: home, end, left, right
-                            (e.keyCode >= 35 && e.keyCode <= 39)) {
-                        // let it happen, don't do anything
-                        return;
-                    }
-                    // Ensure that it is a number and stop the keypress
-                    if ((e.shiftKey || (e.keyCode < 48 || e.keyCode > 57)) && (e.keyCode < 96 || e.keyCode > 105)) {
-                        e.preventDefault();
-                    }
-                });
 
-                function secureSubmitFormHandler() {
-                    var securesubmit_public_key = "<?php echo $publicKey ?>";
+<!-- The SecureSubmit Javascript Library -->
+<script type="text/javascript" src="https://api2.heartlandportico.com/SecureSubmit.v1/token/2.1/securesubmit.js"></script>
+<!-- The Integration Code -->
+<script type="text/javascript">
+    var secureSubmitKey = '<?php echo $publicKey ?>';
 
-                    if ($('input.securesubmitToken').size() === 0) {
-                        var card  = $('#input-cc-number').val().replace(/\D/g, '');
-                        var cvc   = $('#input-cc-cvv2').val();
-                        var exp   = $('#input-cc-expire-date').val().split(' / ');
-                        var month = exp[0];
-                        var year  = exp[1];
-                        (new Heartland.HPS({
-                            publicKey: securesubmit_public_key,
-                            cardNumber: card,
-                            cardCvv: cvc,
-                            cardExpMonth: month,
-                            cardExpYear: year,
-                            success: secureSubmitResponseHandler,
-                            error: secureSubmitResponseHandler
-                        })).tokenize();
-                        return false;
-                    }
+    //call method untill Heartland loaded
+    var interval = setInterval(loadIframeTokenization, 100);
+    var attempts = 0;
 
-                    return true;
+    //method to load iframe
+    function loadIframeTokenization() {
+        'use strict';
+        //check whether Heartland is loaded
+        if (typeof Heartland == 'object') {
+            clearInterval(interval);
+        } else {
+            attempts ++;
+            if(attempts > 20){
+                alert('Problem loading payment Method! Try again later.');
+                clearInterval(interval);
+            }
+            return;
+        }
```

Comment:
I'm not a fan of using an interval here. If we need to ensure that our hosted script is loaded for slow network connections, I'd much rather set the `onload` attribute on the hosted script's tag to call `loadIframeTokenization`. We did this in the OpenCart 2.2 version: [`securesubmit.tpl`](https://github.com/hps/heartland-opencart-2.2-plugin/blob/master/catalog/view/theme/default/template/payment/securesubmit.tpl#L164-L179).

Reply:
To help reduce delays in loading the external script, you can also instruct browsers to preload/prefetch the DNS for our hosted script. See the [OpenCart 2.2 `securesubmit.tpl` file](https://github.com/hps/heartland-opencart-2.2-plugin/blob/master/catalog/view/theme/default/template/payment/securesubmit.tpl#L11-L14) for reference. Just be sure to list the correct domains for the URL you have in place for the hosted script.

Reply:
removed the set interval code